Sdk 869/resume state not working

bklastaitis-branch opened the pull request:
Both onNewIntent and onResume events get fired when doing intra-app linking, and because we call initSession from onDeviceResume in Cordova, simply setting `intent.putExtra("branch_force_new_session", true);` is enough to force a new session.

codinronan commented on src/android/io/branch/BranchSDK.java:
```diff
@@ -298,17 +296,7 @@ private void initSession(CallbackContext callbackContext) {
-
-        this.activity = this.cordova.getActivity();
-        this.instance.reInitSession(this.activity, this.sessionListener);
+        this.instance.initSession(new SessionListener(callbackContext), data, activity);
```
We should also be able to get rid of the private variable `sessionListener` online 56 now right? It's no longer used anywhere since it was only used to gate `reInitSession`.

bklastaitis-branch commented:
> This one is kind of a facepalm one
Yes and no, after core Android SDK v4.2.0, we introduced "SDK already initialized error", so we needed reInitSession, but then we added "force_new_session" intent extra in `onNewIntent` of `BranchSDK.java`, which achieves the same thing as `reInitSession` when used together with `initSession`
